watertap/tools/parallel/parallel_manager.py

```diff
@@ -0,0 +1,150 @@
+from abc import abstractmethod, ABC
+import os
+
+import numpy
+
+from pyomo.common.dependencies import attempt_import
+
+mpi4py, mpi4py_available = attempt_import("mpi4py")
+
+
+class ParallelManager(ABC):
+    @classmethod
+    def create_parallel_manager(cls, parallel_manager_class=None):
+        """
+        Create and return an instance of a ParallelManager, based on the libraries available in the
+        runtime environment.
+
+        Allows an optional python class to be passed in as parallel_manager_class. If so, this class
+        is instantiated and returned rather than checking the local environment.
+        """
+        if parallel_manager_class is not None:
+            return parallel_manager_class()
+
+        if mpi4py_available and ParallelManager.is_running_under_mpi():
+            return MPIParallelManager(mpi4py)
+
+        return SingleProcessParallelManager()
+
+    @classmethod
+    def is_running_under_mpi(cls):
+        """
+        Returns whether the process was run as part of an MPI group.
+        """
+        # see https://www.open-mpi.org/faq/?category=running#mpi-environmental-variables.
+        # this is one of the environment variables that MPI sets.
+        return os.getenv("OMPI_COMM_WORLD_SIZE") is not None
```

Comment:
```suggestion
        from mpi4py import MPI
        return MPI.COMM_WORLD.Get_size() > 1
```
This is not MPI-implementation dependent, and if we're just running in MPI with a single rank we can just use the `SingleProcessParallelManager` anyways.

Reply:
Updated in the latest commit! Good catch.